src/devices/bus/nabupc/adapter.cpp

```diff
@@ -352,45 +511,55 @@ void network_adapter::channel_request(uint8_t byte)
 	++m_substate;
 }
 
-void network_adapter::segment_request(uint8_t byte)
+void network_adapter_base::segment_request(uint8_t byte)
 {
+	static uint32_t segment_id = 0;
+
```

Comment:
A static function variable like this is completely unacceptable.  You should *know* we’re trying to get rid of the remaining static function variables.  Adding them is a regression.